components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp

```diff
@@ -290,8 +290,8 @@
 
         identityProviderClaims = identityProvider.getClaimConfig().getIdpClaims();
 
-        userIdClaimURI = identityProvider.getClaimConfig().getUserClaimURI();
-        roleClaimURI = identityProvider.getClaimConfig().getRoleClaimURI();
+        userIdClaimURI = Encode.forHtmlContent(identityProvider.getClaimConfig().getUserClaimURI());
```

Comment:
Do not encode and store the encoded value in the variable. Do the encoding at the time the value is displayed in the page. Because, encoding should be different based on the place where it is displayed (like element value, element attribute, hyperlink URL etc.)

Reply:
For example, userIdClaimURI is added to the page source as below.

if ('<%=userIdClaimURI%>' == '') {

Above is inside a javascript block. So the correct fix is below.

if ('<%=Encode.forJavaScript(userIdClaimURI)%>' == '') {

Reply:
Fixed in 1e2222e9efdc0ad2025ad5dec052fcc71bfb3bd4